client/utils/events.ts

```diff
@@ -461,14 +464,73 @@ function canMarkEventAsComplete(
     }
 }
 
+/**
+ * Will return true if there is at least one event that can be added
+ */
+export function canAddSomeEventsAsRelatedToPlanningEditor(eventIds: Array<IEventItem['_id']>): boolean {
+    const editor = planningApi.editor(EDITOR_TYPE.INLINE);
+
+    if (editor.manager == null) {
+        return false;
+    }
+
+    const contentProfileHasRelatedEventsField = editor.dom.fields['associated_event'] != null;
+    const planningItem = editor.form.getDiff<IPlanningItem>() as IPlanningItem;
+    const currentRelatedEvents: Array<IPlanningRelatedEventLink> = (planningItem?.related_events ?? []);
+
+    const isEventAlreadyAddedAsRelated: (eventId: string) => boolean =
+        (eventId: string) => currentRelatedEvents.find((item) => item._id === eventId) != null;
+
+    return editor.item.getItemType() === 'planning'
+        && (editor.item.getItemAction() === 'edit' || editor.item.getItemAction() === 'create')
+        && contentProfileHasRelatedEventsField
+        && eventIds.some((eventId) => isEventAlreadyAddedAsRelated(eventId) !== true);
+}
+
+/**
+ * Events that are already added will be ignored
+ */
+export function addSomeEventsAsRelatedToPlanningEditor(eventIds: Array<IEventItem['_id']>) {
+    const editor = planningApi.editor(EDITOR_TYPE.INLINE);
+    const planningItem = editor.form.getDiff<IPlanningItem>() as IPlanningItem;
+
+    const currentRelatedEvents: Array<IPlanningRelatedEventLink> = planningItem?.related_events ?? [];
+
+    const isEventAlreadyAddedAsRelated: (eventId: string) => boolean =
+        (eventId: string) => currentRelatedEvents.find((item) => item._id === eventId) != null;
+
+    const nextRelatedEvents: Array<IPlanningRelatedEventLink> = [
+        ...currentRelatedEvents,
+        ...eventIds
+            .filter((eventId) => isEventAlreadyAddedAsRelated(eventId) !== true)
+            .map((eventId): IPlanningRelatedEventLink => ({
+                _id: eventId,
+                link_type: 'secondary',
```

Comment:
And same here, `link_type` would depend on the config and currently linked Events (if any)